pkg/api/types.go

```diff
@@ -822,6 +822,10 @@ type PodSpec struct {
 	// used must be specified.
 	// Optional: Default to false.
 	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// PullSecretNames is an optional list of secret names to use for pulling any of the images used by this PodSpec.
+	// If specified, these secrets will be passed to individual puller implementations for them to use.  For example,
+	// in the case of docker, only DockerConfig type secrets are honored.
+	PullSecretNames []string `json:"pullSecretNames,omitempty" description:"list of secret names available for pulling the container images"`
```

Comment:
The ServiceAccount PR  lets you decide to not use service accounts (by omitting the admission plugin) and just manage pods/secrets yourself. API tokens, pull secrets, and security context restrictions seem like things that would be valuable to group under a service account, but I'm not sure about requiring service accounts to use those things.